src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,465 @@
+		 * @return this builder
+		 */
+		public Builder withResourceTypeString(final String resourceType) {
+			if (resourceType != null) {
+				try {
+					this.resourceType = ResourceType.getResourceType(resourceType);
+				} catch (IllegalArgumentException e) {
+					// TEMPORARY HACK
+					// As there is only one resource type at present
+					// (ResourceType.DATASET), null input in the resourceType
+					// field will be automatically set to that value.
+					// The error message "resourceType cannot be null or
+					// whitespace only" thus needs to be edited to say
+					// "resourceType cannot be whitespace only".
+					String errorMessage = e.getMessage();
+					if ("resourceType cannot be null or whitespace only"
+							.equals(errorMessage)) {
+						errorMessage = "resourceType cannot be whitespace only";
+					}
+					errorList.add(errorMessage);
+				}
+			}
+			else {
+				this.resourceType = DEFAULT_RESOURCE_TYPE;
+			}
+			return this;
+		}
+
+		/**
+		 * Sets the resource type for the resource.
+		 *
+		 * As there is currently only one resource type, entering null will set the
+		 * resource type value to DEFAULT_RESOURCE_TYPE in the resultant object.
+		 *
+		 * @param resourceType
+		 *                resource type as a {@link ResourceType}
+		 * @return this builder
+		 */
+		public Builder withResourceType(final ResourceType resourceType) {
+			// TEMPORARY HACK - only one resource type at present
+			this.resourceType = resourceType == null ? DEFAULT_RESOURCE_TYPE : resourceType;
+			return this;
+		}
+
+		/**
+		 * Sets comments for the resource.
+		 *
+		 * @param comments
+		 *                comments as a list of strings
+		 * @return this builder
+		 */
+		public Builder withComments(final List<String> comments) {
+			if (comments != null) {
+				this.comments = comments.stream()
+						.filter(Objects::nonNull)
+						.map(String::trim)
+						.filter(c -> !Util.isNullOrWhitespace(c))
+						.distinct()
+						.collect(Collectors.toList());
+			}
+			else {
+				this.comments = comments;
+			}
+			return this;
+		}
+
+		/**
+		 * Sets the license for the resource; can be a string like 'Apache 2.0' or an
+		 * URL-like string; URL-like strings will be checked for well-formedness.
+		 * For these purposes, a license string that starts with a series of letters
+		 * followed by one or more colons and one or more slashes (e.g. "https://") is
+		 * considered URL-like.
+		 *
+		 * @param license
+		 *                license as a string
+		 * @return this builder
+		 */
+		public Builder withLicense(final String license) {
+			this.license = Common.processString(license);
+			return this;
+		}
+
+		/**
+		 * Sets the version.
+		 *
+		 * Valid credit metadata objects require either a version or one or more dates.
+		 *
+		 * @param withVersion
+		 *                version string
+		 * @return this builder
+		 */
+		public Builder withVersion(final String version) {
+			this.version = Common.processString(version);
```

Comment:
Do you want to have a maximum length for the version (and I guess any of these fields)? We'll definitely need a check for the overall size of the CM regardless

Reply:
I will retrofit sizes if it seems like it's something we should add.

Reply:
:+1: